lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -144,49 +225,17 @@ export abstract class CacheManager implements ICacheManager {
                 }
             });
         }
-        this.saveCredential(credential);
-    }
-
-    /**
-     * retrieve an account entity given the cache key
-     * @param key
-     */
-    getAccount(key: string): AccountEntity | null {
-        // don't parse any non-account type cache entities
-        if (CredentialEntity.getCredentialType(key) !== Constants.NOT_DEFINED || this.isAppMetadata(key)) {
-            return null;
-        }
-
-        // Attempt retrieval
-        let entity: AccountEntity;
-        try {
-            entity = this.getItem(key, CacheSchemaType.ACCOUNT) as AccountEntity;
-        } catch (e) {
-            return null;
-        }
-
-        // Authority type is required for accounts, return if it is not available (not an account entity)
-        if (!entity || StringUtils.isEmpty(entity.authorityType)) {
-            return null;
-        }
-
-        return entity;
-    }
-
-    /**
-     * retrieve a credential - accessToken, idToken or refreshToken; given the cache key
-     * @param key
-     */
-    getCredential(key: string): CredentialEntity | null {
-        return this.getItem(key, CacheSchemaType.CREDENTIAL) as CredentialEntity;
+        const key = credential.generateCredentialKey();
+        this.setAccessTokenCredential(key, credential);
     }
 
     /**
-     * retrieve an appmetadata entity given the cache key
-     * @param key
+     * saves accessToken credential
+     * @param credential
      */
-    getAppMetadata(key: string): AppMetadataEntity | null {
-        return this.getItem(key, CacheSchemaType.APP_METADATA) as AppMetadataEntity;
+    private saveRefreshToken(credential: RefreshTokenEntity): void {
+        const key = credential.generateCredentialKey();
```

Comment:
Would moving the key generation to `setRefreshTokenCredential` work?

Reply:
Yes and Done.